central/policy/service/service_impl.go

```diff
@@ -401,6 +407,22 @@ func (s *serviceImpl) CancelDryRunJob(ctx context.Context, jobid *v1.JobId) (*v1
 	return &v1.Empty{}, nil
 }
 
+func (s *serviceImpl) getNetworkPoliciesForDeployment(ctx context.Context, dep *storage.Deployment) (*augmentedobjs.NetworkPoliciesApplied, error) {
+	storedPolicies, err := s.networkPolicies.GetNetworkPolicies(ctx, dep.GetClusterId(), dep.GetNamespace())
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get network policies for clusterId %s on namespace %s", dep.GetClusterId(), dep.GetNamespace())
+	}
+
+	matchedNetworkPolicies := map[string]*storage.NetworkPolicy{}
+	for _, p := range storedPolicies {
+		if labels.MatchLabels(p.GetSpec().GetPodSelector(), dep.GetPodLabels()) {
+			matchedNetworkPolicies[p.GetId()] = p
+		}
+	}
+
+	return networkpolicy.GetNetworkPoliciesApplied(matchedNetworkPolicies), nil
+}
+
```

Comment:
+!, looks like a good util function. Not sure if the imports would allow it but `pkg` would be a good place as well (removing the datastore access and passing in the stored policies)

Reply:
Done